x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/components/search_ui_form.tsx

```diff
@@ -0,0 +1,205 @@
+  { defaultMessage: 'Title field (Optional)' }
+);
+const TITLE_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.titleFieldHelpText',
+  { defaultMessage: 'Used as the top-level visual identifier for every rendered result' }
+);
+const TITLE_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.titleFieldPlaceholder',
+  { defaultMessage: 'Select a title field' }
+);
+const FILTER_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.filterFieldLabel',
+  { defaultMessage: 'Filter fields (Optional)' }
+);
+const FILTER_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.filterFieldHelpText',
+  { defaultMessage: 'Faceted values rendered as filters and available as query refinement' }
+);
+const FILTER_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.filterFieldPlaceholder',
+  { defaultMessage: 'Click to select' }
+);
+const SORT_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortFieldLabel',
+  { defaultMessage: 'Sort fields (Optional)' }
+);
+const SORT_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortHelpText',
+  { defaultMessage: 'Used to display result sorting options, ascending and descending' }
+);
+const SORT_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortFieldPlaceholder',
+  { defaultMessage: 'Click to select' }
+);
+const URL_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldLabel',
+  { defaultMessage: 'URL field (Optional)' }
+);
+const URL_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldHelpText',
+  { defaultMessage: "Used as a result's link target, if applicable" }
+);
+const URL_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldPlaceholder',
+  { defaultMessage: 'Select a URL field' }
+);
+const GENERATE_PREVIEW_BUTTON_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.generatePreviewButtonLabel',
+  { defaultMessage: 'Generate a Preview' }
+);
+
+interface Option<T> extends Omit<EuiComboBoxOptionOption<T>, 'value'> {
+  value: string;
+}
+
+export const SearchUIForm: React.FC = () => {
+  const {
+    validFields,
+    validSortFields,
+    validFacetFields,
+    titleField,
+    urlField,
+    facetFields,
+    sortFields,
+  } = useValues(SearchUILogic);
+  const { engineName } = useValues(EngineLogic);
+  const {
+    onActiveFieldChange,
+    onFacetFieldsChange,
+    onSortFieldsChange,
+    onTitleFieldChange,
+    onUrlFieldChange,
+  } = useActions(SearchUILogic);
+
+  const previewHref = generatePreviwUrl(
+    {
+      fromKibana: 'true',
+      titleField,
+      urlField,
+      facets: facetFields,
+      sortFields,
+    },
+    engineName
+  );
```

Comment:
Sorry, I probably should have grouped all my `generatePreviewUrl` util comments in a single pass, apologies they're all over the place!

In addition to passing `fromKibana` by default in the helper (rather than here), I also think the helper should be in charge of fetching `engineName` itself, rather than passing it as a arg. Example:

```js
import { EngineLogic } from '../../engine';

export const generatePreviewUrl = (queries) => {
  const { engineName } = EngineLogic.values;

  return queryString.stringifyUrl(
    {
      query: { ...queries, fromKibana: true },
      url: getAppSearchUrl(`/engines/${engineName}/reference_application/preview`),
    },
    // ...etc
```

And API usage would then simplify greatly to this:

```js
  const previewHref = generatePreviewUrl({
    titleField,
    urlField,
    facets: facetFields,
    sortFields,
  });
```